Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Components/Operations/Wizard/SpeckleOperationWizard.cs

```diff
@@ -0,0 +1,324 @@
+﻿using System.Diagnostics;
+using Microsoft.Extensions.DependencyInjection;
+using Speckle.Connectors.Common.Operations;
+using Speckle.Connectors.GrasshopperShared.HostApp;
+using Speckle.Connectors.GrasshopperShared.Registration;
+using Speckle.Sdk;
+using Speckle.Sdk.Api;
+using Speckle.Sdk.Api.GraphQL.Inputs;
+using Speckle.Sdk.Api.GraphQL.Models;
+using Speckle.Sdk.Credentials;
+using Version = Speckle.Sdk.Api.GraphQL.Models.Version;
+
+namespace Speckle.Connectors.GrasshopperShared.Components.Operations.Wizard;
+
+/// <summary>
+/// Wizard to handle cascading selections in an order Workspace, Project, Model and Version for operations.
+/// Wraps the UI components with it and exposes the state of the selection to consumer.
+/// </summary>
+public class SpeckleOperationWizard
+{
+  private readonly IClientFactory _clientFactory;
+
+  public Account? SelectedAccount { get; private set; }
+  public List<Account> Accounts { get; private set; }
+  public Workspace? SelectedWorkspace { get; private set; }
+  public Project? SelectedProject { get; private set; }
+  public Model? SelectedModel { get; private set; }
+  public Version? SelectedVersion { get; private set; }
+
+  public WorkspaceMenuHandler WorkspaceMenuHandler { get; }
+  public ProjectMenuHandler ProjectMenuHandler { get; }
+  public ModelMenuHandler ModelMenuHandler { get; }
+  public VersionMenuHandler? VersionMenuHandler { get; }
+
+  private readonly Func<Task> _refreshComponent;
+  private readonly AccountService _accountService;
+  private readonly AccountManager _accountManager;
+
+  /// <param name="refreshComponent"> Callback function to trigger when component need to refresh itself.</param>
+  /// <param name="isSender"> Whether it will be used in sender or receiver. Accordingly, the wizard will manage versions or not.</param>
+  public SpeckleOperationWizard(Func<Task> refreshComponent, bool isSender)
+  {
+    _refreshComponent = refreshComponent;
+    _clientFactory = PriorityLoader.Container.GetRequiredService<IClientFactory>();
+    _accountManager = PriorityLoader.Container.GetRequiredService<AccountManager>();
+    _accountService = PriorityLoader.Container.GetRequiredService<AccountService>();
+
+    var userSelectedAccountId = _accountService.GetUserSelectedAccountId();
+    SelectedAccount =
+      userSelectedAccountId != null
+        ? _accountManager.GetAccount(userSelectedAccountId)
+        : _accountManager.GetDefaultAccount();
```

Comment:
Let's try to not use the default here, shouldn't be null at this point but throw exception if it is